### DISCUSSION TITLE
Sandboxing and restriction?

### DISCUSSION THREAD

**Discussion by jerome-trc** [fubark/cyber#97] [General]
I'm eyeing using Cyber for game engine modding but would only do so if there are ways to prevent third-party scripts from acting as malware (accessing the file system, reaching out to the internet, dealing with raw pointers, FFI, executing system commands, etc.). I'm not currently seeing any options to do this.

Is this on the roadmap, and I should submit a PR? Or is it something I should customise? Thanks.

**Comment by fubark** [comment]
For embedding Cyber, restricting the API is already possible. You would be responsible for providing the modules that user scripts can import and use. Then the only thing left is disallowing the use of pointers in user scripts which is planned. I think it should be introduced as a VM flag. "safe" or "sandbox" Then sema would return an error when encountering pointer semantics.

**Reply by jerome-trc** [reply]
I see, I misunderstood the section of the documentation on the standard modules. Indeed, along with prohibiting URL imports, I could probably chip in anything I need if/when I commit to Cyber and get the time. Thanks for the help.
